frontend/src/components/Login.tsx

```diff
@@ -38,14 +59,53 @@ const Login = () => {
       <div className="lg:w-1/2 flex items-center justify-center p-8">
         <div className="max-w-md w-full space-y-8">
           <div className="text-center">
-            <h2 className="text-3xl font-bold text-gray-900">Welcome Back</h2>
-            <p className="mt-2 text-gray-600">Please sign in to your account</p>
+            {signUp ? (
+              <div>
+                <h2 className="text-3xl font-bold text-gray-900">Welcome..!</h2>
+                <p className="mt-2 text-gray-600">Please Create your account</p>
+              </div>
+            ) : (
+              <div>
+                <h2 className="text-3xl font-bold text-gray-900">
+                  Welcome Back..!
+                </h2>
+                <p className="mt-2 text-gray-600">
+                  Please sign in to your account
+                </p>
+              </div>
+            )}
           </div>
 
           <form onSubmit={handleSubmit} className="mt-8 space-y-6">
             <div className="space-y-4">
+              {signUp ? (
+                <div>
+                  <label
+                    htmlFor="email"
+                    className="block text-start font-bold text-sm  text-gray-700"
+                  >
+                    Username
+                  </label>
+                  <input
+                    id="username"
+                    name="username"
+                    type="text"
+                    required
+                    value={formData.username}
+                    onChange={handleChange}
+                    className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm focus:outline-none focus:ring-blue-500 focus:border-blue-500"
+                    placeholder="Enter your email"
```

Comment:
add a UI library or use styled-components lib for custom components with that you can reuse it by importing